code/modules/integrated_electronics/subtypes/weaponized.dm

```diff
@@ -0,0 +1,344 @@
+	The 'fire' activator will cause the mechanism to attempt to launch objects at the coordinates, if possible. Note that the \
+	projectile needs to be inside the machine, or on an adjacent tile, and must be medium sized or smaller. The assembly \
+	must also be a gun if you wish to launch something while the assembly is in hand."
+	complexity = 75
+	w_class = WEIGHT_CLASS_SMALL
+	size = 4
+	cooldown_per_use = 30
+	ext_cooldown = 15
+	inputs = list(
+		"target X rel" = IC_PINTYPE_NUMBER,
+		"target Y rel" = IC_PINTYPE_NUMBER,
+		"projectile" = IC_PINTYPE_REF,
+		"canister" = IC_PINTYPE_REF
+		)
+	outputs = list()
+	activators = list(
+		"fire" = IC_PINTYPE_PULSE_IN
+	)
+	spawn_flags = IC_SPAWN_RESEARCH
+	action_flags = IC_ACTION_COMBAT
+	power_draw_per_use = 50
+	var/mole_requirement = 1
+
+/obj/item/integrated_circuit/weaponized/air_cannon/do_work()
+
+	var/max_w_class = assembly.w_class
+	var/target_x_rel = round(get_pin_data(IC_INPUT, 1))
+	var/target_y_rel = round(get_pin_data(IC_INPUT, 2))
+	var/obj/item/A = get_pin_data_as_type(IC_INPUT, 3, /obj/item)
+	var/obj/item/integrated_circuit/atmospherics/AT = get_pin_data_as_type(IC_INPUT, 4, /obj/item/integrated_circuit/atmospherics)
+
+	if(!A || A.anchored || A.throwing || A == assembly || istype(A, /obj/item/twohanded) || istype(A, /obj/item/transfer_valve))
+		return
+
+	if(!AT || !AT.air_contents)
+		return
+
+	if (istype(assembly.loc, /obj/item/implant/storage)) //Prevents the more abusive form of chestgun.
+		return
+
+	if(max_w_class && (A.w_class > max_w_class))
+		return
+
+	if(!assembly.can_fire_equipped && ishuman(assembly.loc))
+		return
+
+	// Is the target inside the assembly or close to it?
+	if(!check_target(A, exclude_components = TRUE))
+		return
+
+	var/turf/T = get_turf(get_object())
+	if(!T)
+		return
+
+	// If the item is in mob's inventory, try to remove it from there.
+	if(ismob(A.loc))
+		var/mob/living/M = A.loc
+		if(!M.temporarilyRemoveItemFromInventory(A))
+			return
+
+	var/datum/gas_mixture/source_air = AT.return_air()
+	var/datum/gas_mixture/target_air = T.return_air()
+
+	if(!source_air || !target_air)
+		return
+
+	if(source_air.total_moles() <= mole_requirement)
+		return
+
+	// If the item is in a grabber circuit we'll update the grabber's outputs after we've thrown it.
+	var/obj/item/integrated_circuit/manipulation/grabber/G = A.loc
+
+	var/x_abs = CLAMP(T.x + target_x_rel, 0, world.maxx)
+	var/y_abs = CLAMP(T.y + target_y_rel, 0, world.maxy)
+	var/range = round(CLAMP(sqrt(target_x_rel*target_x_rel+target_y_rel*target_y_rel),0,8),1)
+	assembly.visible_message("<span class='danger'>[assembly] has thrown [A]!</span>")
+	log_attack("[assembly] [REF(assembly)] has thrown [A] with lethal force.")
+	A.forceMove(drop_location())
+	A.throw_at(locate(x_abs, y_abs, T.z), range, 3)
+
+	var/datum/gas_mixture/removed = source_air.remove(mole_requirement)
```

Comment:
Wouldnt it be better to make it be based off KPA instead? It makes no logical sense for a pneumatic cannon to function off the number of moles of gas instead of the KPA

Reply:
The pneumatic cannon uses moles as a requirement to shoot things. I'm just going to copy it because I don't know how to tell it to take the kPa of something.

Reply:
*scream